dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/OkHttpUtils.java

```diff
@@ -35,53 +39,148 @@
+     * @throws RuntimeException
+     */
+    public static @NonNull OkHttpResponse get(@NonNull String url,
+                                              @Nullable OkHttpRequestHeaders okHttpRequestHeaders,
+                                              @Nullable Map<String, Object> requestParams,
+                                              int connectTimeout,
+                                              int writeTimeout,
+                                              int readTimeout) throws IOException {
+        CLIENT.newBuilder()
+                .connectTimeout(connectTimeout, TimeUnit.MILLISECONDS)
+                .writeTimeout(writeTimeout, TimeUnit.MILLISECONDS)
+                .readTimeout(readTimeout, TimeUnit.MILLISECONDS)
+                .build();
         String finalUrl = addUrlParams(requestParams, url);
         Request.Builder requestBuilder = new Request.Builder().url(finalUrl);
-        addHeader(httpHeaders, requestBuilder);
+        addHeader(okHttpRequestHeaders.getHeaders(), requestBuilder);
         Request request = requestBuilder.build();
         try (Response response = CLIENT.newCall(request).execute()) {
-            return getResponseBody(response);
+            return new OkHttpResponse(response.code(), getResponseBody(response));
+        } catch (Exception e) {
+            throw new RuntimeException(String.format("Get request execute failed, url: %s", url), e);
         }
     }
 
-    public static @NonNull String post(@NonNull String url,
-                                       @Nullable Map<String, String> httpHeaders,
-                                       @Nullable Map<String, Object> requestParamsMap,
-                                       @Nullable Map<String, Object> requestBodyMap) throws IOException {
+    /**
+     * http post request
+     * @param connectTimeout connect timeout in milliseconds
+     * @param writeTimeout write timeout in milliseconds
+     * @param readTimeout read timeout in milliseconds
+     * @return OkHttpResponse
+     * @throws RuntimeException
+     */
+    public static @NonNull OkHttpResponse post(@NonNull String url,
+                                               @Nullable OkHttpRequestHeaders okHttpRequestHeaders,
+                                               @Nullable Map<String, Object> requestParamsMap,
+                                               @Nullable Map<String, Object> requestBodyMap,
+                                               int connectTimeout,
+                                               int writeTimeout,
+                                               int readTimeout) throws IOException {
+        CLIENT.newBuilder()
+                .connectTimeout(connectTimeout, TimeUnit.MILLISECONDS)
+                .writeTimeout(writeTimeout, TimeUnit.MILLISECONDS)
+                .readTimeout(readTimeout, TimeUnit.MILLISECONDS)
+                .build();
         String finalUrl = addUrlParams(requestParamsMap, url);
```

Comment:
Use FormBody is better than directly add the params to url.
https://square.github.io/okhttp/recipes/#__tabbed_7_2

Reply:
`FormBody` doesn't apply to get request. I think we can keep it that way.